src/main/java/microsoft/exchange/webservices/data/ExchangeServiceBase.java

```diff
@@ -471,28 +479,35 @@ protected Date convertUniversalDateTimeStringToDate(String dateString) {
     if (dateString == null || dateString.isEmpty()) {
       return null;
     } else {
-      if (dateString.endsWith("Z")) {
+    TimeZone UTCTimeZone = TimeZone.getTimeZone("UTC");
```

Comment:
I also would suggest in adding more comments to the things being done. This can be usefull if someone else will look at the code later on.